Userland/Libraries/LibWeb/HTML/HTMLTableElement.cpp

```diff
@@ -104,18 +111,24 @@ JS::GCPtr<HTMLTableSectionElement> HTMLTableElement::t_head()
     return nullptr;
 }
 
+// https://html.spec.whatwg.org/multipage/tables.html#dom-table-thead
 WebIDL::ExceptionOr<void> HTMLTableElement::set_t_head(HTMLTableSectionElement* thead)
 {
-    // FIXME: This is not always the case, but this function is currently written in a way that assumes non-null.
-    VERIFY(thead);
-
-    if (thead->local_name() != TagNames::thead)
+    // If the new value is neither null nor a thead element, then a "HierarchyRequestError" DOMException must be thrown instead.
+    if (thead && thead->local_name() != TagNames::thead)
         return WebIDL::HierarchyRequestError::create(realm(), "Element is not thead");
 
-    // FIXME: The spec requires deleting the current thead if thead is null
-    //        Currently the wrapper generator doesn't send us a nullable value
+    // On setting, if the new value is null or a thead element, the first thead element child of the table element,
+    // if any, must be removed,
     delete_t_head();
 
+    if (!thead)
+        return {};
+
+    // and the new value, if not null, must be inserted immediately before the first element in the table element
+    // that is neither a caption element nor a colgroup element, if any,
+    // or at the end of the table if there are no such elements.
+
     // We insert the new thead after any <caption> or <colgroup> elements
     DOM::Node* child_to_append_after = nullptr;
```

Comment:
Shouldn't this variable be called `child_to_append_before`?
The `<thead>` element is inserted after any `<caption>` or `<colgroup>` element, but the variable points to the first element that is neither of those. The `<thead>` element is inserted before this child, as per [pre-insert](https://dom.spec.whatwg.org/#concept-node-pre-insert) specifications.